test/project_types/script/layers/infrastructure/assemblyscript_task_runner_test.rb

```diff
@@ -61,15 +61,68 @@
   describe ".dependencies_installed?" do
     subject { as_task_runner.dependencies_installed? }
 
-    it "should return true if node_modules folder exists" do
+    before do
       FileUtils.mkdir_p("node_modules")
+    end
+
+    it "should return true if node_modules folder exists" do
+      stub_npm_outdated({})
       assert_equal true, subject
     end
 
     it "should return false if node_modules folder does not exists" do
       Dir.stubs(:exist?).returns(false)
+      stub_npm_outdated({})
       assert_equal false, subject
     end
+
+    it "should not error if `npm outdated` returns nothing" do
+      stub_npm_outdated({})
+      subject
+    end
+
+    it "should not error if `npm outdated` does not return an EP package" do
+      stub_npm_outdated(create_package_version_info(package_name: "somepackage"))
+      subject
+    end
+
+    it "should not error if current version is linked" do
+      stub_npm_outdated(create_package_version_info(current: "linked"))
+      subject
+    end
+
+    it "should not error if latest version is an https URL" do
+      stub_npm_outdated(create_package_version_info(latest: "https://github.com/somethingsomething"))
+      subject
+    end
+
+    it "should not error if patch version is different" do
+      stub_npm_outdated(create_package_version_info(current: "0.9.0", latest: "0.9.1"))
+      subject
+    end
+
+    it "should not error if it's a non-zero major version and minor version is different" do
+      stub_npm_outdated(create_package_version_info(current: "1.0.0", latest: "1.1.0"))
+      subject
+    end
+
+    it "should error if it's a zero major version and minor version is different" do
+      package_name = "@shopify/extension-point-as-foo"
+      stub_npm_outdated(create_package_version_info(package_name: package_name, current: "0.9.0", latest: "0.10.0"))
+      msg = "The following packages must be updated before you can push: #{package_name}"
+      assert_raises Script::Layers::Infrastructure::Errors::PackagesOutdatedError, msg do
+        subject
+      end
```

Comment:
Does this test case pass?  How come I see the `msg` is different from what's raised?
```
raise Errors::PackagesOutdatedError.new(outdated_ep_packages),
            "NPM packages out of date: #{outdated_ep_packages.join(', ')}" unless outdated_ep_packages.empty?
```

Reply:
It does pass but yeah, the message doesn't appear to be validated like I expected. I'm going to look into it.

Reply:
IIRC passing a message in `assert_raises` just mocks that as the exception object's message

Reply:
Yeah I updated the code. One thing that's weird is when there's a mismatch, we get an error message that says:

```
No visible difference in the String#inspect output.
You should look at the implementation of #== on String or its members.
```

Which is weird since they're both strings.